sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -443,6 +443,26 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val FILE_STREAM_SINK_LOG_DELETE = SQLConfigBuilder("spark.sql.streaming.fileSink.log.deletion")
+    .internal()
+    .doc("Whether to delete the expired log files in file stream sink.")
+    .booleanConf
+    .createWithDefault(true)
+
+  val FILE_STREAM_SINK_LOG_COMPACT_LEN =
+    SQLConfigBuilder("spark.sql.streaming.fileSink.log.compactLen")
+      .internal()
+      .doc("Every how many log files is a compaction triggered.")
+      .intConf
+      .createWithDefault(10)
+
+  val FILE_STREAM_SINK_LOG_EXPIRED_TIME =
+    SQLConfigBuilder("spark.sql.streaming.fileSink.log.expired")
+      .internal()
+      .doc("How long in milliseconds a file is guaranteed to be visible for all readers.")
+      .longConf
```

Comment:
This should be a `timeConf` isnt it ? Then any one can use "1 second", "1 minute".